src/power_grid_model_io/utils/uuid_excel_cvtr.py

```diff
@@ -0,0 +1,176 @@
+#
+# SPDX-License-Identifier: MPL-2.0
+
+"""
+    This is a stand-alone conversion script that converts the GUID based Vision 
+    excel files to a number based format. The script is designed to be used as 
+    a light weight tool to run before running the PGM-IO conversion scripts.
+    Example usage:
+
+    convert_guid_vision_excel("vision_97_en.xlsx", number="Number")
+    convert_guid_vision_excel("vision_97_nl.xlsx", number="Nummer")
+
+"""
+
+import re
+from typing import Optional
+
+import pandas as pd
+
+
+class UUID2IntCvtr:
+    """
+    Class for bookkeeping the conversion of GUIDs to integers
+    """
+
+    def __init__(self, uuids: Optional[list] = None) -> None:
+        """Initialize with a list of UUIDs, or empty list
+
+        Args:
+            uuids (list, optional): GUID's in Vision excel. Defaults to [].
+        """
+        if uuids is None:
+            uuids = []
+        self._uuids_int: dict[str, int] = {}
+        self._counter: int = 0
+        for uuid in set(uuids):
+            self.add(uuid)
+
+    def add_list(self, uuids: list) -> None:
+        """Add a list of UUIDs
+
+        Args:
+            uuids (list): GUID's in Vision excel
+        """
+        for uuid in set(uuids):
+            self.add(uuid)
+
+    def add(self, uuid: str) -> None:
+        """Add a single entry of UUID
+
+        Args:
+            uuid (str): A single GUID in Vision excel
+        """
+        if uuid not in self._uuids_int:
+            self._uuids_int[uuid] = self._counter
+            self._counter += 1
+
+    def query(self, uuid: str) -> int:
+        """Get the singular integer value respective of a UUID input
+
+        Args:
+            uuid (str): A single GUID in Vision excel
+
+        Returns:
+            str: the singular integer value
+        """
+        try:
+            return self._uuids_int[uuid]
+        except KeyError:
+            return -1
+
+    def get_keys(self) -> list:
+        """Get the keys in the dictionary
+
+        Returns:
+            list: the whole list of keys
+        """
+        return list(self._uuids_int.keys())
+
+    def get_size(self) -> int:
+        """Get how many valid UUIDs are in the dictionary
+
+        Returns:
+            int: the size of the dictionary
+        """
+        return self._counter
+
+
+def load_excel_file(file_name: str) -> pd.ExcelFile:
+    """Load an excel file
+
+    Args:
+        file_name (str): Excel file name
+
+    Returns:
+        pd.ExcelFile: pandas ExcelFile object
+    """
+    return pd.ExcelFile(file_name)
+
+
+def get_guid_columns(df: pd.DataFrame) -> list:
+    """Get the columns that contain the word "GUID"
+
+    Args:
+        df (pd.DataFrame): panda dataframe
+
+    Returns:
+        list: list of columns containing the word "GUID"
+    """
+    guid_regex = re.compile(r".*GUID")
+    return df.filter(regex=guid_regex).columns
```

Comment:
is there a guarantee that no accidental processions of the form `r".*GUID.*"` are matched?

if not, we need to be smarter about that.

if so, pandas documentation says it uses `re.search` behavior under the hood. In that case, just matching for `r"GUID"` should suffice combined with the explanation in the docstring

Reply:
Went with `(r".*GUID$")`